### PR TITLE
Fixing broken installer 

### LINKED ISSUE #1107: Installer fails: error: Setup script exited with 'pkg-config' not found
**To Reproduce**
1. Clone main repo and run installer with -fiap
2. See error

**LibreTime version:**
Latest commit

**Installation method:**
 - OS: Ubuntu Cloud Image
 - OS Version: Bionic
 - Method: install -fiap

**Screenshots**
```
Installed /usr/local/lib/python3.6/dist-packages/pika-1.1.0-py3.6.egg
Searching for PyGObject>=3.34.0
Reading https://pypi.org/simple/PyGObject/
Downloading https://files.pythonhosted.org/packages/3a/a7/de282a4aaedba59d60a895a7821e6497b39cbdfa94a352776ff45ffc6e6f/PyGObject-3.38.0.tar.gz#sha256=051b950f509f2e9f125add96c1493bde987c527f7a0c15a1f7b69d6d1c3cd8e6
Best match: PyGObject 3.38.0
Processing PyGObject-3.38.0.tar.gz
Writing /tmp/easy_install-n5mc2apo/PyGObject-3.38.0/setup.cfg
Running PyGObject-3.38.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-n5mc2apo/PyGObject-3.38.0/egg-dist-tmp-xd6_j2y0
error: Setup script exited with 'pkg-config' not found.
Arguments: ('pkg-config', '--print-errors', '--exists', 'glib-2.0 >= 2.48.0')

Try installing it with: 'sudo apt install pkg-config'
```

**Additional context**
I see that Travis CI failed to build the latest commit... :(

### PR DESCRIPTION
**Pull request by zklosko**
This PR fixes an installer crash on Ubuntu 18.04 (Bionic) after failing to install PyGObject. Fixes #1107; possibly fixes #1092. Here's the band-aid code:

```
# Manual installation of pip3 components to make PyGObject work
echo "Manual installation of pip3 components to make PyGObject work"
localUser=$(who am i | awk '{print $1}')
loudCmd sudo -u $localUser pip3 install pycairo
loudCmd sudo -u $localUser pip3 install PyGObject
```

### CODE REVIEW

File: install
Comment (paddatrapper) [inline install:757]: This should be part of the `setup.py` file for the python app that this breaks. The dependencies should be listed with the others there

Comment (paddatrapper) [reply install:757]: By the looks of the errors, these should be in `python_apps/airtime_analyzer/setup.py`

Comment (zklosko) [reply install]: When putting it in the way you mentioned, I encounter the same error as last time.

Also, about the dependency issue, one of the PyGObject maintainers writes:
>py3cairo is the pkg-config name, not related to pypi.
Seems you are missing pycairo, but pip should install it before since it's an install dependency, so not sure how this can happen.
Did you disable isolated builds in pip?

File: python_apps/airtime_analyzer/setup.py
Comment (paddatrapper) [inline python_apps/airtime_analyzer/setup.py:42]: The trailing `,` is helpful for anyone un-commenting the other dependencies

File: install
Comment (paddatrapper) [reply install]: So then they require a user owned version of those dependencies...? That is very strange and potentially breaks things (e.g. in a container when the user is root

File: python_apps/airtime_analyzer/setup.py
Comment (zklosko) [reply python_apps/airtime_analyzer/setup.py:42]: Ah, okay. I thought it would cause issues.

File: install
Comment (zklosko) [reply install]: No idea.

**Comment by zklosko** [comment]
```
Searching for PyGObject==3.34.0
Reading https://pypi.org/simple/PyGObject/
Downloading https://files.pythonhosted.org/packages/46/8a/b183f3edc812d4d28c8b671a922b5bc2863be5d38c56b3ad9155815e78dd/PyGObject-3.34.0.tar.gz#sha256=2acb0daf2b3a23a90f52066cc23d1053339fee2f5f7f4275f8baa3704ae0c543
Best match: PyGObject 3.34.0
Processing PyGObject-3.34.0.tar.gz
Writing /tmp/easy_install-4hb50ecw/PyGObject-3.34.0/setup.cfg
Running PyGObject-3.34.0/setup.py -q bdist_egg --dist-dir /tmp/easy_install-4hb50ecw/PyGObject-3.34.0/egg-dist-tmp-5yn_s7f7
Package py3cairo was not found in the pkg-config search path.
Perhaps you should add the directory containing `py3cairo.pc'
to the PKG_CONFIG_PATH environment variable
No package 'py3cairo' found
Traceback (most recent call last):
  File "/tmp/easy_install-4hb50ecw/PyGObject-3.34.0/setup.py", line 178, in _run_pkg_config
  File "/usr/lib/python3.6/subprocess.py", line 356, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '('pkg-config', '--print-errors', '--exists', 'py3cairo >= 1.11.1')' returned non-zero exit status 1.
```

Still no dice. Should we just apply the band-aid for now until a better fix can be put together?

**Comment by paddatrapper** [comment]
Under vagrant the work-around fails with the following error:

    ubuntu-bionic: Manual installation of pip3 components to make PyGObject work
    ubuntu-bionic: sudo -u pip3 install pycairo
    ubuntu-bionic: sudo
    ubuntu-bionic: : 
    ubuntu-bionic: unknown user: pip3
    ubuntu-bionic: sudo
    ubuntu-bionic: : 
    ubuntu-bionic: unable to initialize policy plugin

**Comment by zklosko** [comment]
I'm not sure exactly how Vagrant works, but the previous line gets the username that's currently logged in/is passing the instructions to sudo. That's how the `-u` flag is used. Does Vagrant not have users?

**Comment by paddatrapper** [comment]
Vagrant has users, logging into a running VM gives me a `vagrant` user ssh session. I believe that the provisioning scripts (e.g. `install`) are run as root

**Comment by zklosko** [comment]
```
localUser=$(who am i | awk '{print $1}')
loudCmd sudo -u $localUser pip3 install pycairo
```

This code should run `pip3 install pycairo` under the *vagrant* user. 😢 

Going back to Python dependencies, are there any other methods we haven't tried?

**Comment by paddatrapper** [comment]
https://tracker.debian.org/pkg/pygobject

It seems that pygobject and its cairo bindings are packaged for Debian and Ubuntu. We could try use those instead...

**Comment by zklosko** [comment]
For Debian, yes; said packages are not in Ubuntu's repos.

**Comment by paddatrapper** [comment]
They're in Ubuntu: https://launchpad.net/ubuntu/+source/pygobject

**Comment by zklosko** [comment]
Hmm... `apt install` can't find them. Do you know what the repo is?

**Comment by paddatrapper** [comment]
python3-gi and python3-gi-cairo should be the package names. Either in main or universe

**Comment by zklosko** [comment]
Those packages were already being installed. Still doesn't work.

**Comment by Robbt** [comment]
Just looking around this seems like an annoying bug. I haven't had a chance to try to get the install to work with this but are we installing all three of these packages "sudo apt install python3-gi python3-gi-cairo gir1.2-gtk-3.0" ? - I just pulled that from - https://pygobject.readthedocs.io/en/latest/getting_started.html#ubuntu-getting-started which was mentioned as a solution.

**Comment by zklosko** [comment]
@Robbt Yeah, all of that's being installed 😢

**Comment by zklosko** [comment]
Also, would it be better to not have Composer run as root? https://getcomposer.org/doc/faqs/how-to-install-untrusted-packages-safely.md

**Comment by Robbt** [comment]
Hmm, looks like 18.04 uses 3.26.1-2 and above it says it is looking for PyGObject>=3.34.0 so that could be the problem. Python package management has always been a source of various headaches due to evolving version numbers and the static nature of LibreTime. Do we know what package triggered this new dependency on the newer version of PyGObject and have we tried forcing an older version of that library ?

In some ways it might make sense to have the version hard coded and set for each python package so that it can install on older distros like 18.04.

**Comment by zklosko** [comment]
I just noticed the Python packages get installed in reverse, compared to how they're listed in the array in *setup.py*. I'm starting to think **mutagen** is the problem because every other package installs correctly before the crash.

Update: rolling back to the following dependencies makes the installer work

```
install_requires=[
          'mutagen==1.42.0',
          'pika~=1.1.0',
          'file-magic',
          'nose',
          'coverage',
          'mock',
          'python-daemon',
          'requests>=2.7.0',
          'rgain3==1.0.0',
          'pycairo==1.19.1',
          'PyGObject==3.26.1',
```

**Comment by zklosko** [comment]
Also, can we update Travis CI to do the test builds on Bionic instead of Xenial since Xenial is losing LTS support soon?

**Comment by crudexec** [comment]
> Also, can we update Travis CI to do the test builds on Bionic instead of Xenial since Xenial is losing LTS support soon?
> 
> Do we use Selenium as well for testing?



> I just noticed the Python packages get installed in reverse, compared to how they're listed in the array in _setup.py_. I'm starting to think **mutagen** is the problem because every other package installs correctly before the crash.
> 
> Update: rolling back to the following dependencies makes the installer work
> 
> ```
> install_requires=[
>           'mutagen==1.42.0',
>           'pika~=1.1.0',
>           'file-magic',
>           'nose',
>           'coverage',
>           'mock',
>           'python-daemon',
>           'requests>=2.7.0',
>           'rgain3==1.0.0',
>           'pycairo==1.19.1',
>           'PyGObject==3.26.1',
> ```

Fantastic, should I wait for this to get merged in or can I build from this commit hash?

**Comment by zklosko** [comment]
@crudexec you can, but I'd wait for input from @Robbt and @paddatrapper

**Comment by paddatrapper** [comment]
Do we know exactly what broke that the pinning fixes? I.e. how do we know when we can upgrade those dependencies to newer versions?

**Comment by Robbt** [comment]
If I remember correctly we previously had mutagen pinned and then we decided to let it update/install newer versions to try to fix some issues. I could be wrong but I think it would be wise to just pin these dependencies for now. I suspect based upon what we are seeing and the bundled versions of python packages that we might be able to unpin them when we get 20.04 working.

 Which seems unlikely to happen soon unless we fork Propel to apply the fixes or otherwise convince the maintainers to apply the patches that make it compatible. I dont think Propel 2.0 is the solution as far as I can tell because it was never fully released. Moving away from Propel would involve a major rewrite as well.

**Comment by paddatrapper** [comment]
I've added the required packages to the other supported distros and busy testing in Vagrant

**Comment by paddatrapper** [comment]
Debian Buster errors:

    debian-buster: Searching for PyGObject==3.26.1
    debian-buster: Reading https://pypi.org/simple/PyGObject/
    debian-buster: No local packages or working download links found for PyGObject==3.26.1
    debian-buster: error: Could not find suitable distribution for Requirement.parse('PyGObject==3.26.1')

**Comment by paddatrapper** [comment]
There is a 3.36.1 version. Is that the one we need?

**Comment by paddatrapper** [comment]
With 3.36.1, I run into the same issue as before:

    debian-buster: Running PyGObject-3.36.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-r1auuhad/PyGObject-3.36.1/egg-dist-tmp-j29aa59h
    debian-buster: Package py3cairo was not found in the pkg-config search path.
    debian-buster: Perhaps you should add the directory containing `py3cairo.pc'                                                                                                               
    debian-buster: to the PKG_CONFIG_PATH environment variable                                                                                                                                 
    debian-buster: No package 'py3cairo' found
    debian-buster: Traceback (most recent call last):
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 178, in _run_pkg_config                                                                                 
    debian-buster:   File "/usr/lib/python3.7/subprocess.py", line 395, in check_output                                                                                                        
    debian-buster:     **kwargs).stdout                                                                                                                                                        
    debian-buster:   File "/usr/lib/python3.7/subprocess.py", line 487, in run                                                                                                                 
    debian-buster:     output=stdout, stderr=stderr)                                                                                                                                           
    debian-buster: subprocess.CalledProcessError: Command '('pkg-config', '--print-errors', '--exists', 'py3cairo >= 1.11.1')' returned non-zero exit status 1.                                
    debian-buster:                                                                                                                                                                             
    debian-buster: During handling of the above exception, another exception occurred:                                                                                                         
    debian-buster:                                                                                                                                                                             
    debian-buster: Traceback (most recent call last):                                                                                                                                          
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 186, in _run_pkg_config                                                                                 
    debian-buster:   File "/usr/lib/python3.7/subprocess.py", line 395, in check_output                                                                                                        
    debian-buster:     **kwargs).stdout                                                                                                                                                        
    debian-buster:   File "/usr/lib/python3.7/subprocess.py", line 487, in run                                                                                                                 
    debian-buster:     output=stdout, stderr=stderr)                                                                                                                                           
    debian-buster: subprocess.CalledProcessError: Command '['pkg-config', '--exists', 'py3cairo']' returned non-zero exit status 1.                                                            
    debian-buster:                                                                                                                                                                             
    debian-buster: During handling of the above exception, another exception occurred:                                                                                                         
    debian-buster:                                                                                                                                                                             
    debian-buster: Traceback (most recent call last):                                                                                                                                          
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 199, in _run_pkg_config_or_exit                                                                         
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 188, in _run_pkg_config                                                                                 
    debian-buster: __main__.PkgConfigMissingPackageError: Command '('pkg-config', '--print-errors', '--exists', 'py3cairo >= 1.11.1')' returned non-zero exit status 1.                        
    debian-buster:                                                                                                                                                                             
    debian-buster: During handling of the above exception, another exception occurred:                                                                                                         
    debian-buster:                                                                                                                                                                             
    debian-buster: Traceback (most recent call last):                                                                                                                                          
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 152, in save_modules                                                                            
    debian-buster:     yield saved                                                                                                                                                             
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 193, in setup_context                                                                           
    debian-buster:     yield                                                                                                                                                                   
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 254, in run_setup                                                                               
    debian-buster:     _execfile(setup_script, ns)                                                                                                                                             
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 43, in _execfile                                                                                
    debian-buster:     exec(code, globals, locals)                                                                                                                                             
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1288, in <module>                                                                                       
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1283, in main                                                                                           
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/__init__.py", line 153, in setup                                                                                  
    debian-buster:     return distutils.core.setup(**attrs)                                                                                                                                    
    debian-buster:   File "/usr/lib/python3.7/distutils/core.py", line 148, in setup                                                                                                           
    debian-buster:     dist.run_commands()                                                                                                                                                     
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 966, in run_commands                                                                                                    
    debian-buster:     self.run_command(cmd)                                                                                                                                                   
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command                                                                                                     
    debian-buster:     cmd_obj.run()                                                                                                                                                           
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/bdist_egg.py", line 167, in run                                                                           
    debian-buster:     cmd = self.call_command('install_lib', warn_dir=0)                                                                                                                      
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/bdist_egg.py", line 153, in call_command                                                                  
    debian-buster:     self.run_command(cmdname)                                                                                                                                               
    debian-buster:   File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command                                                                                                      
    debian-buster:     self.distribution.run_command(command)                                                                                                                                  
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command                                                                                                     
    debian-buster:     cmd_obj.run()                                                                                                                                                           
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/install_lib.py", line 11, in run                                                                          
    debian-buster:     self.build()                                                                                                                                                            
    debian-buster:   File "/usr/lib/python3.7/distutils/command/install_lib.py", line 109, in build                                                                                            
    debian-buster:     self.run_command('build_ext')                                                                                                                                           
    debian-buster:   File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command                                                                                                      
    debian-buster:     self.distribution.run_command(command)                                                                                                                                  
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command                                                                                                     
    debian-buster:     cmd_obj.run()                                                                                                                                                           
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1118, in run                                                                                            
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1110, in _setup_extensions                                                                              
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1093, in add_pycairo                                                                                    
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 927, in get_pycairo_include_dir                                                                         
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 911, in find_pkg_config                                                                                 
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 222, in pkg_config_version_check                                                                        
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 213, in _run_pkg_config_or_exit                                                                         
    debian-buster: SystemExit: Command '('pkg-config', '--print-errors', '--exists', 'py3cairo >= 1.11.1')' returned non-zero exit status 1.                                                   
    debian-buster:                                                                                                                                                                             
    debian-buster: During handling of the above exception, another exception occurred:                                                                                                         
    debian-buster:                                                                                                                                                                             
    debian-buster: Traceback (most recent call last):                                                                                                                                          
    debian-buster:   File "/vagrant/python_apps/airtime_analyzer/setup.py", line 50, in <module>                                                                                               
    debian-buster:     data_files=data_files)                                                                                                                                                  
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/__init__.py", line 153, in setup                                                                                  
    debian-buster:     return distutils.core.setup(**attrs)                                                                                                                                    
    debian-buster:   File "/usr/lib/python3.7/distutils/core.py", line 148, in setup                                                                                                           
    debian-buster:     dist.run_commands()                                                                                                                                                     
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 966, in run_commands                                                                                                    
    debian-buster:     self.run_command(cmd)                                                                                                                                                   
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command                                                                                                     
    debian-buster:     cmd_obj.run()                                                                                                                                                           
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/install.py", line 67, in run                                                                              
    debian-buster:     self.do_egg_install()                                                                                                                                                   
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/install.py", line 117, in do_egg_install                                                                  
    debian-buster:     cmd.run(show_deprecation=False)                                                                                                                                         
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 408, in run                                                                        
    debian-buster:     self.easy_install(spec, not self.no_deps)                                                                                                                               
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 650, in easy_install                                                               
    debian-buster:     return self.install_item(None, spec, tmpdir, deps, True)                                                                                                                
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 697, in install_item                                                               
    debian-buster:     self.process_distribution(spec, dist, deps)                                                                                                                             
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 742, in process_distribution                                                       
    debian-buster:     [requirement], self.local_index, self.easy_install                                                                                                                      
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py", line 766, in resolve                                                                             
    debian-buster:     replace_conflicting=replace_conflicting                                                                                                                                 
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py", line 1049, in best_match                                                                         
    debian-buster:     return self.obtain(req, installer)                                                                                                                                      
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/pkg_resources/__init__.py", line 1061, in obtain                                                                             
    debian-buster:     return installer(requirement)                                                                                                                                           
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 669, in easy_install                                                               
    debian-buster:     return self.install_item(spec, dist.location, tmpdir, deps)                                                                                                             
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 695, in install_item                                                               
    debian-buster:     dists = self.install_eggs(spec, download, tmpdir)                                                                                                                       
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 880, in install_eggs                                                               
    debian-buster:     return self.build_and_install(setup_script, setup_base)                                                                                                                 
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 1150, in build_and_install                                                         
    debian-buster:     self.run_setup(setup_script, setup_base, args)                                                                                                                          
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/easy_install.py", line 1134, in run_setup                                                                 
    debian-buster:     run_setup(setup_script, args)                                                                                                                                           
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 257, in run_setup                                                                               
    debian-buster:     raise                                                                                                                                                                   
    debian-buster:   File "/usr/lib/python3.7/contextlib.py", line 130, in __exit__                                                                                                            
    debian-buster:     self.gen.throw(type, value, traceback)                                                                                                                                  
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 193, in setup_context                                                                           
    debian-buster:     yield                                                                                                                                                                   
    debian-buster:   File "/usr/lib/python3.7/contextlib.py", line 130, in __exit__                                                                                                            
    debian-buster:     self.gen.throw(type, value, traceback)                                                                                                                                  
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 164, in save_modules                                                                            
    debian-buster:     saved_exc.resume()                                                                                                                                                      
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 139, in resume                                                                                  
    debian-buster:     raise exc.with_traceback(self._tb)                                                                                                                                      
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 152, in save_modules                                                                            
    debian-buster:     yield saved                                                                                                                                                             
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 193, in setup_context                                                                           
    debian-buster:     yield                                                                                                                                                                   
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 254, in run_setup                                                                               
    debian-buster:     _execfile(setup_script, ns)                                                                                                                                             
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/sandbox.py", line 43, in _execfile                                                                                
    debian-buster:     exec(code, globals, locals)                                                                                                                                             
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1288, in <module>
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1283, in main
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/__init__.py", line 153, in setup
    debian-buster:     return distutils.core.setup(**attrs)
    debian-buster:   File "/usr/lib/python3.7/distutils/core.py", line 148, in setup
    debian-buster:     dist.run_commands()
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 966, in run_commands
    debian-buster:     self.run_command(cmd)
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    debian-buster:     cmd_obj.run()
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/bdist_egg.py", line 167, in run
    debian-buster:     cmd = self.call_command('install_lib', warn_dir=0)
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/bdist_egg.py", line 153, in call_command
    debian-buster:     self.run_command(cmdname)
    debian-buster:   File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command
    debian-buster:     self.distribution.run_command(command)
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    debian-buster:     cmd_obj.run()
    debian-buster:   File "/usr/local/lib/python3.7/dist-packages/setuptools/command/install_lib.py", line 11, in run
    debian-buster:     self.build()
    debian-buster:   File "/usr/lib/python3.7/distutils/command/install_lib.py", line 109, in build
    debian-buster:     self.run_command('build_ext')
    debian-buster:   File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command
    debian-buster:     self.distribution.run_command(command)
    debian-buster:   File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    debian-buster:     cmd_obj.run()
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1118, in run
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1110, in _setup_extensions
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 1093, in add_pycairo
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 927, in get_pycairo_include_dir
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 911, in find_pkg_config
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 222, in pkg_config_version_check
    debian-buster:   File "/tmp/easy_install-r1auuhad/PyGObject-3.36.1/setup.py", line 213, in _run_pkg_config_or_exit
    debian-buster: setuptools.sandbox.UnpickleableException: SystemExit(PkgConfigMissingPackageError(CalledProcessError(1, ('pkg-config', '--print-errors', '--exists', 'py3cairo >= 1.11.1')))

**Comment by zklosko** [comment]
Is there a repo we can add that would make the packages install on all OSes?

**Comment by paddatrapper** [comment]
It may be possible to fix using <= 3.36.1 as the version requirement in setup.py

**Comment by zklosko** [comment]
@paddatrapper I think I tried that; it wouldn't work past 3.26.1 on Ubuntu 18.04

**Comment by paddatrapper** [comment]
> 3.26.1

That version doesn't exist according to PyPi - https://pypi.org/project/PyGObject/#history

**Comment by paddatrapper** [comment]
Bionic fails with the same error as Buster

**Comment by paddatrapper** [comment]
> It may be possible to fix using <= 3.36.1 as the version requirement in setup.py

Works on Buster

**Comment by zklosko** [comment]
@paddatrapper Do we still use Selenium?

**Comment by paddatrapper** [comment]
> @paddatrapper Do we still use Selenium?

Not that I know of

**Comment by Robbt** [comment]
> > 3.26.1
> 
> That version doesn't exist according to PyPi - https://pypi.org/project/PyGObject/#history

3.26  is probably just too old for them to keep a copy of it on PyPi their archives stop at 2017, evidently it is the one that bionic uses it, which means we can install it from the debian package in Ubuntu.

Selenium has not been implemented as far as I know since the project transitioned to LibreTime.

**Comment by zklosko** [comment]
> > It may be possible to fix using <= 3.36.1 as the version requirement in setup.py
> 
> Works on Buster

Works on Ubuntu Bionic too!

**Comment by zklosko** [comment]
@paddatrapper @Robbt I was also curious about Travis CI: do we need all of the tests it's running and can we update it to run tests on Ubuntu Bionic? Travis CI is currently averaging a 4,000 job backlog and so it might benefit us to not run unnecessary tests...

**Comment by paddatrapper** [comment]
@zklosko shall we move this discussion to chat.libretime.org so that we don't spam this PR?

**Comment by Robbt** [comment]
Travis CI is something I've only messed with on occasion. We probably only need to test Travis CI on one distro in general in terms of determining whether code changes break. It may be useful to do some basic test to determine if a distro just breaks but also people seem to be pretty good at reporting that. So we would probably be fine if we only ran one test. I haven't done much with the Travis CI in general and our test suite has never been particularly thorough but mostly legacy from tests developed when the project was part of SourceFabric and had multiple full time developers.

**Comment by paddatrapper** [comment]
Works on Xenial too. So <= 3.36.1 works for all versions we support

**Comment by zklosko** [comment]
@Robbt Currently we're testing for PHP 5.6, 7.0, 7.1, 7.2, 7.3, and Python 3.5, 3.7, 3.8. Are any of those unnecessary?

From what I remember, Xenial uses PHP 7.0 and Bionic uses PHP 7.2; no idea for Python.

**Comment by zklosko** [comment]
@paddatrapper Can we quickly finish up here? I think we're almost done.

**Comment by paddatrapper** [comment]
The issue is that PHP 7.2 causes test failures (#779) that we cannot fix without replacing Zend. The versions should reflect the platforms we support (Centos, Debian Buster, Ubuntu Bionic, Ubuntu Xenial). Ideally we should also test against whatever the latest version is too. So if someone wants to try work out what those versions are, we can evaluate which versions are not necessary.

**Comment by zklosko** [comment]
@Robbt @paddatrapper Back to the installer: are we feeling comfortable to merge as is?

**Comment by Robbt** [comment]
I haven't tested it but if it works with Buster and 18.04 I think we should merge it.

**Comment by crudexec** [comment]
> I haven't tested it but if it works with Buster and 18.04 I think we should merge it.

Just wanted to mention that I have been following this PR since it was opened majorly because I need to install Libretime for my project, but I appreciate the effort you guys put into making free software work well. looking forward to the merge !

**Comment by paddatrapper** [comment]
Two things before we merge: please do not remove the selenium tests as part of this PR. That should be separate. Also I want to test that things actually run with this change (doing that now).

**Comment by paddatrapper** [comment]
Thanks. Testing looks good. Merging

**Review by paddatrapper — APPROVED** [review]

**Pull request merged**
